src/client.ts

```diff
@@ -27,80 +28,244 @@ import type { Browsers } from "./types.ts";
+   */
+  #browser_process_closed = false;
+
+  /**
+   * What browser are we running?
+   */
+  readonly browser: Browsers;
+
+  /**
+   * The collection of page objects for a user to interact with
+   */
+  #pages: Page[] = [];
+
+  /**
+   * Only if the browser is firefox, is this present.
+   * This is the path to the directory that firefox uses
+   * to write a profile
+   */
+  readonly #firefox_profile_path?: string;
+
+  /**
+   * The host and port that the websocket server is listening on
+   */
+  readonly wsOptions: {
+    hostname: string;
+    port: number;
+  };
+
+  /**
+   * @param protocol - The browser protocol to interact with
+   * @param browserProcess - The browser process to interact with
+   * @param browser - The name of the browser we will be running
+   * @param wsOptions - The debugger options
+   * @param firefoxProfilePath - The path to the firefox dev profile (if applicable)
+   */
   constructor(
     protocol: ProtocolClass,
+    browserProcess: Deno.Process,
+    browser: Browsers,
+    wsOptions: {
+      hostname: string;
+      port: number;
+    },
+    firefoxProfilePath?: string,
   ) {
     this.#protocol = protocol;
+    this.#browser_process = browserProcess;
+    this.browser = browser;
+    this.wsOptions = wsOptions;
+    this.#firefox_profile_path = firefoxProfilePath;
   }
 
   /**
-   * Close/stop the sub process, and close the ws connection. Must be called when finished with all your testing
+   * Only for internal use. No documentation or help
+   * will be provided to users regarding this method
+   *
+   * This was only created so we could make `pages` property private,
+   * but still allow the Page class to remove a page from the list
+   *
+   * @param pageTargetId - Target id of the page to remove
    */
-  public async done() {
-    await this.#protocol.done();
+  public _popPage(pageTargetId: string) {
+    this.#pages = this.#pages.filter((page) => page.target_id !== pageTargetId);
   }
 
   /**
-   * Go to a specific page
+   * For internal use.
    *
-   * @param urlToVisit - The page to go to
+   * Pushed a new item to the pages array
    *
-   * @returns A page instance, with methods to help you directly interact with the page
+   * @param page - Page to push
    */
-  public async goTo(urlToVisit: string): Promise<Page> {
-    const method = "Page.loadEventFired";
-    this.#protocol.notification_resolvables.set(method, deferred());
-    const notificationPromise = this.#protocol.notification_resolvables.get(
-      method,
-    );
-    const res = await this.#protocol.sendWebSocketMessage<
-      ProtocolTypes.Page.NavigateRequest,
-      ProtocolTypes.Page.NavigateResponse
-    >(
-      "Page.navigate",
-      {
-        url: urlToVisit,
-      },
-    );
-    await notificationPromise;
-    if (res.errorText) {
-      await this.#protocol.done(
-        `${res.errorText}: Error for navigating to page "${urlToVisit}"`,
+  public _pushPage(
```

Comment:
is the `_` to denote "for internal use" ?

Reply:
yeah

Reply:
alright cool. resolving